jupyter_sphinx/__init__.py

```diff
@@ -1 +1,256 @@
+    app.add_node(
+        JupyterKernelNode,
+        html=(skip, None),
+        latex=(skip, None),
+        textinfo=(skip, None),
+        text=(skip, None),
+        man=(skip, None),
+    )
+
+    # JupyterCellNode is a container that holds the input and
+    # any output, so we render it as a container.
+    app.add_node(
+        JupyterCellNode,
+        html=render_container,
+        latex=render_container,
+        textinfo=render_container,
+        text=render_container,
+        man=render_container,
+    )
+
+    # JupyterWidgetViewNode holds widget view JSON,
+    # but is only rendered properly in HTML documents.
+    app.add_node(
+        JupyterWidgetViewNode,
+        html=(visit_element_html, None),
+        latex=(skip, None),
+        textinfo=(skip, None),
+        text=(skip, None),
+        man=(skip, None),
+    )
+    # JupyterWidgetStateNode holds the widget state JSON,
+    # but is only rendered in HTML documents.
+    app.add_node(
+        JupyterWidgetStateNode,
+        html=(visit_element_html, None),
+        latex=(skip, None),
+        textinfo=(skip, None),
+        text=(skip, None),
+        man=(skip, None),
+    )
+
+    # ThebeSourceNode holds the source code and is rendered if
+    # hide-code is not specified. For HTML it is always rendered,
+    # but hidden using the stylesheet
+    app.add_node(
+        ThebeSourceNode,
+        html=(visit_container_html, depart_container_html),
+        latex=render_thebe_source,
+        textinfo=render_thebe_source,
+        text=render_thebe_source,
+        man=render_thebe_source,
+    )
+
+    # ThebeOutputNode holds the output of the Jupyter cells
+    # and is rendered if hide-output is not specified.
+    app.add_node(
+        ThebeOutputNode,
+        html=(visit_container_html, depart_container_html),
+        latex=render_container,
+        textinfo=render_container,
+        text=render_container,
+        man=render_container,
+    )
+
+    # ThebeButtonNode is the button that activates thebelab
+    # and is only rendered for the HTML builder
+    app.add_node(
+        ThebeButtonNode,
+        html=(visit_element_html, None),
+        latex=(skip, None),
+        textinfo=(skip, None),
+        text=(skip, None),
+        man=(skip, None),
+    )
+
+    app.add_directive("jupyter-execute", JupyterCell)
+    app.add_directive("jupyter-kernel", JupyterKernel)
+    app.add_directive("thebe-button", ThebeButton)
+    app.add_role("jupyter-download:notebook", jupyter_download_role)
+    app.add_role("jupyter-download:script", jupyter_download_role)
+    app.add_transform(ExecuteJupyterCells)
+
+    # For syntax highlighting
+    app.add_lexer("ipythontb", IPythonTracebackLexer())
+    app.add_lexer("ipython", IPython3Lexer())
+
+    app.connect("builder-inited", builder_inited)
+    app.connect("build-finished", build_finished)
+
+    return {"version": __version__, "parallel_read_safe": True}
+
+
+def setup(app):
```

Comment:
This seems to be identical to `setup = _setup`; are you sure that we cannot just call `__init__.setup` from `execute`?

Reply:
ah yeah - I just refactored a bit more so something like this is done

Reply:
Thanks!